### PR TITLE
Local Deployment + Location management + HEALTH connectivity + bug fixes

### PR DESCRIPTION
**Pull request by bardram**
This PR :

 * Supports local deployment using environment variables (#211, #125)
 * Improves location management / permissions (#223)
 * Integrates the health sampling package (#221, #220)
 * Fixes a set of small issues (#225, #157, #222)

Also updated API + README documentation in places.

### CODE REVIEW

File: lib/carp_study_app.dart [hunk 1]
```diff
@@ -45,13 +46,22 @@ class CarpStudyAppState extends State<CarpStudyApp> {
...
-              redirect: (context, state) => (!bloc.backend.isAuthenticated)
-                  ? LoginPage.route
-                  : (!bloc.hasStudyBeenDeployed)
-                      ? InvitationListPage.route
-                      : (!bloc.hasInformedConsentBeenAccepted)
-                          ? InformedConsentPage.route
-                          : firstRoute),
+              redirect: (context, state) {
+                if (bloc.deploymentMode != DeploymentMode.local) {
+                  // check authentication to CAWS
```
Comment (LarsRefsgaard) [inline lib/carp_study_app.dart:51]: i'd say remove unnecessary comments, it's implicit in the logic

File: lib/carp_study_app.dart [hunk 2]
```diff
@@ -118,9 +128,9 @@ class CarpStudyAppState extends State<CarpStudyApp> {
       GoRoute(
         path: InformedConsentPage.route,
         parentNavigatorKey: _rootNavigatorKey,
-        redirect: (context, state) => bloc.hasInformedConsentBeenAccepted
-            ? firstRoute
-            : (bloc.studyId == null ? InvitationListPage.route : null),
+        // redirect: (context, state) => bloc.hasInformedConsentBeenAccepted
+        //     ? firstRoute
+        //     : (bloc.studyId == null ? InvitationListPage.route : null),
```
Comment (LarsRefsgaard) [inline lib/carp_study_app.dart:133]: what happens if the user gets redirected to the ic page through deep linking for example, do they have to accept it again?

File: lib/main.dart [hunk 3]
```diff
@@ -40,14 +40,17 @@ import 'package:carp_esense_package/esense.dart';
 import 'package:carp_polar_package/carp_polar_package.dart';
 import 'package:research_package/research_package.dart';
 import 'package:cognition_package/cognition_package.dart';
+import 'package:carp_health_package/health_package.dart';
+// import 'package:health/health.dart';
```
Comment (LarsRefsgaard) [inline lib/main.dart:44]: remove dead code comments

File: pubspec.yaml [hunk 4]
```diff
@@ -46,7 +46,7 @@ dependencies:
   go_router: ^13.0.0
   flutter_svg: ^2.0.4
   flutter_blue_plus: ^1.15.5
-  intl: ^0.18.1
+  intl: '>=0.18.0 <1.0.0'
```
Comment (LarsRefsgaard) [inline pubspec.yaml:49]: are you sure about this?

**Comment by LarsRefsgaard** [comment]
xcode is not building with these commits... and also gitguardian is complaining about exposed secrets, these need to be remedied

**Comment by LarsRefsgaard** [comment]
maybe next time it would be beneficial to keep PRs on a smaller scope so there's not 2500 code changes to look through...

**Comment by Panosfunk** [comment]
> xcode is not building with these commits... and also gitguardian is complaining about exposed secrets, these need to be remedied

I had the same issue last week(uploaded secrets). I think its better if we changed the history of the commits too so that the keys do not appear in old commits. 

I fixed this using [this link](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository#using-git-filter-repo). It's a tool that automates a lot of the process but I still needed to do some things on my own.

**Comment by bardram** [comment]
> maybe next time it would be beneficial to keep PRs on a smaller scope so there's not 2500 code changes to look through...

I agree - but the problem is that often you "touch" many files just in order to make a "small" change..

File: pubspec.yaml (hunk 4 above)
Comment (bardram) [reply pubspec.yaml:49]: Yes - there has been some discussion on the `intl` package and problem with Flutter - not solved yet, so this is the current work-around.

File: lib/carp_study_app.dart (hunk 2 above)
Comment (bardram) [reply lib/carp_study_app.dart:133]: I would say yes - the reason to navigate to the IC is to show it to the user (again). This is actually a missing features right now, that the user should be able to go back and see the IC - e.g. from the setting menu.

**Comment by LarsRefsgaard** [comment]
> > maybe next time it would be beneficial to keep PRs on a smaller scope so there's not 2500 code changes to look through...
> 
> I agree - but the problem is that often you "touch" many files just in order to make a "small" change..

Sure, but cannot accept this PR before the app is properly building on iOS.

**Comment by bardram** [comment]
This was included in PR #234

**Pull request closed without merging**